velox/functions/prestosql/Bitwise.h

```diff
@@ -0,0 +1,191 @@
+template <typename T>
+VELOX_UDF_BEGIN(bitwise_left_shift)
+FOLLY_ALWAYS_INLINE bool call(int64_t& result, T number, T shift) {
+  if constexpr (std::is_same<T, int8_t>::value) {
+    return bitwiseLeftShift<T, 8>(result, number, shift);
+  } else if constexpr (std::is_same<T, int16_t>::value) {
+    return bitwiseLeftShift<T, 16>(result, number, shift);
+  } else if constexpr (std::is_same<T, int32_t>::value) {
+    return bitwiseLeftShift<T, 32>(result, number, shift);
+  } else {
+    return bitwiseLeftShift<T, 64>(result, number, shift);
+  }
+}
+VELOX_UDF_END();
+
+template <typename T, typename UT, int MAX_SHIFT>
+FOLLY_ALWAYS_INLINE bool bitwiseRightShift(int64_t& result, T number, T shift) {
+  if ((uint32_t)shift >= MAX_SHIFT) {
+    result = 0;
+  } else {
+    result = ((UT)number >> shift);
+  }
+  return true;
+}
+
+template <typename T>
+VELOX_UDF_BEGIN(bitwise_right_shift)
+FOLLY_ALWAYS_INLINE bool call(int64_t& result, T number, T shift) {
+  if constexpr (std::is_same<T, int8_t>::value) {
+    return bitwiseRightShift<int8_t, uint8_t, 8>(result, number, shift);
+  } else if constexpr (std::is_same<T, int16_t>::value) {
+    return bitwiseRightShift<int16_t, uint16_t, 16>(result, number, shift);
+  } else if constexpr (std::is_same<T, int32_t>::value) {
+    return bitwiseRightShift<int32_t, uint32_t, 32>(result, number, shift);
+  } else {
+    return bitwiseRightShift<int64_t, uint64_t, 64>(result, number, shift);
+  }
+}
+VELOX_UDF_END();
+
+template <typename T, uint64_t MASK, uint64_t SIGNED_BIT>
+FOLLY_ALWAYS_INLINE int64_t preserveSign(T number) {
+  if ((number & SIGNED_BIT) != 0) {
+    return (number | ~MASK);
+  } else {
+    return (number & MASK);
+  }
+}
+
+template <typename T>
+VELOX_UDF_BEGIN(bitwise_right_shift_arithmetic)
+FOLLY_ALWAYS_INLINE bool call(int64_t& result, T number, T shift) {
+  if ((uint32_t)shift >= 64) {
+    if (number >= 0) {
+      result = 0L;
+    } else {
+      result = -1L;
+    }
+    return true;
+  }
+
+  if constexpr (std::is_same<T, int8_t>::value) {
+    result = preserveSign<int8_t, 0b11111111L, 0b10000000L>(number) >> shift;
+  } else if constexpr (std::is_same<T, int16_t>::value) {
+    result = preserveSign<int16_t, 0b1111111111111111L, 0b1000000000000000L>(
+                 number) >>
+        shift;
+  } else if constexpr (std::is_same<T, int32_t>::value) {
+    result = preserveSign<int32_t, 0x00000000ffffffffL, 0x000000000080000000L>(
+                 number) >>
+        shift;
+  } else {
+    result = number >> shift;
+  }
+
+  return true;
+}
+VELOX_UDF_END();
+
+VELOX_UDF_BEGIN(bitwise_logical_shift_right)
+FOLLY_ALWAYS_INLINE bool
+call(int64_t& result, int64_t number, int64_t shift, int64_t bits) {
+  // Presto defines this only for bigint, thus we will define this only for
+  // int64_t.
+  if (bits == 64) {
+    result = number >> shift;
+    return true;
+  }
+
+  VELOX_USER_CHECK(
+      !(bits <= 1 || bits > 64), "Bits specified in must be between 2 and 64");
```

Comment:
nit: drop "specified in" from the message for brevity and include the "wrong" value:  "Bits must be between 2 and 64: {}", bits